src/lib/migration/chainCache.ts

```diff
@@ -0,0 +1,50 @@
+import { ServiceFactory, LocalStorageService } from "../classes";
+import { ChainData } from "../classes/services/chainsService";
+
+const CURRENT_PERSISTED_DATA_VERSION = 2;
+/**
+ *
+ */
+export function checkAndMigrate(): void {
+    const storageService = ServiceFactory.get<LocalStorageService>(LocalStorageService.ServiceName);
+    const version = storageService.load("persisted data version") ?? 1; // Get the current version
+    const shouldMigratePersistedChains = version < CURRENT_PERSISTED_DATA_VERSION;
+    if (shouldMigratePersistedChains) {
+        migrateEachVersion();
+    }
+}
+/**
+ *
+ */
+function migrateEachVersion(): void {
+    const storageService = ServiceFactory.get<LocalStorageService>(LocalStorageService.ServiceName);
+    let version: number = storageService.load("persisted data version") ?? 1; // Get the current version
+    for (CURRENT_PERSISTED_DATA_VERSION; version < CURRENT_PERSISTED_DATA_VERSION; version++) {
+        migratePersistedChains(version + 1);
```

Comment:
I would make this agnostic and keep a map of verison numbers and migrations from vx to vx+1, just like here https://github.com/iotaledger/firefly/blob/develop/packages/shared/lib/core/profile/actions/profiles/checkAndMigrateProfiles.ts#L42. With the code you have now you are forcing every single migration to be the exact same function and logic, just the version that is set at the end changes, but all migrations will do the same